VMEncryption/main/ResourceDiskUtil.py

```diff
@@ -18,134 +18,90 @@
 
 """ Functionality to encrypt the Azure resource disk"""
 
-import uuid
 import time
-import json
-import types
 import os
 
 from CommandExecutor import CommandExecutor
 from Common import CommonVariables
-from DiskUtil import DiskUtil
+
 
 class ResourceDiskUtil(object):
     """ Resource Disk Encryption Utilities """
 
-    RD_KEY_FILE = CommonVariables.PassphraseFileNameKey
     RD_MOUNT_POINT = '/mnt/resource'
     RD_BASE_DEV_PATH = os.path.join(CommonVariables.azure_symlinks_dir, 'resource')
     RD_DEV_PATH = os.path.join(CommonVariables.azure_symlinks_dir, 'resource-part1')
-    DM_PREFIX = CommonVariables.dev_mapper_root
+    DEV_DM_PREFIX = '/dev/dm-'
     # todo: consolidate this and other key file path references
     # (BekUtil.py, ExtensionParameter.py, and dracut patches)
-    RD_KEY_FILE = '/mnt/azure_bek_disk/LinuxPassPhraseFileName'
-    RD_KEY_FILE_MOUNT_POINT = '/mnt/azure_bek_disk'
-    RD_KEY_VOLUME_LABEL = 'BEK VOLUME'
+    RD_MAPPER_NAME = 'resourceencrypt'
+    RD_MAPPER_PATH = os.path.join(CommonVariables.dev_mapper_root, RD_MAPPER_NAME)
 
-    def __init__(self, hutil, logger, distro_patcher):
-        self.hutil = hutil
+    def __init__(self, logger, disk_util, passphrase_filename, public_settings):
         self.logger = logger
         self.executor = CommandExecutor(self.logger)
-        self.disk_util = DiskUtil(hutil=self.hutil, patching=distro_patcher, logger=self.logger, encryption_environment=None)
-        self.mapper_name = str(uuid.uuid4())
-        self.mapper_path = os.path.join(self.DM_PREFIX, self.mapper_name)
+        self.disk_util = disk_util
+        self.passphrase_filename = passphrase_filename  # WARNING: This may be null, in which case we mount the resource disk if its unencrypted and do nothing if it is.
+        self.public_settings = public_settings
 
-    def is_encrypt_format_all(self):
+    def _is_encrypt_format_all(self):
         """ return true if current encryption operation is EncryptFormatAll """
-        try:                
-            public_settings_str = self.hutil._context._config['runtimeSettings'][0]['handlerSettings'].get('publicSettings')
-            if isinstance(public_settings_str, basestring):
-                public_settings = json.loads(public_settings_str)
-            else:
-                public_settings = public_settings_str
-            encryption_operation = public_settings.get(CommonVariables.EncryptionEncryptionOperationKey)
-            if encryption_operation in [CommonVariables.EnableEncryptionFormatAll]:
-                return True
-        except:
-            self.logger.log("unable to identify current encryption operation")
+        encryption_operation = self.public_settings.get(CommonVariables.EncryptionEncryptionOperationKey)
+        if encryption_operation in [CommonVariables.EnableEncryptionFormatAll]:
+            return True
+        self.logger.log("unable to identify current encryption operation")
```

Comment:
Wording of this log message may be confusing to someone reading the log later, since we did pull an encryption operation it just wasn't an encrypt format all operation.   Maybe change to "current encryption operation is not EnableEncryptionFormatAll"

Reply:
Will do.